src/embeddings/embeddings_calculator.cc

```diff
@@ -123,42 +128,76 @@ class EmbeddingsCalculator : public CalculatorBase {
+                size_t input_ids_size = tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape()[1];
+                if (input_ids_size > max_context_length) {
+                    SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Input size {} exceeds max_context_length {}", input_ids_size, max_context_length);
+                    return absl::InvalidArgumentError(absl::StrCat("Input length ", input_ids_size, " longer than allowed ", max_context_length));
+                }
+                for (const auto& embeddingsInputName : embeddingsInputNames) {
+                    auto it = tokenizerOutputMap.find(embeddingsInputName);
+                    RET_CHECK(it != tokenizerOutputMap.end());
+                    SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
+                    embeddingsInputMap[embeddingsInputName] = it->second;
+                    if (embeddingsInputName == "attention_mask") {
+                        if (received_batch_size == 1) {
+                            handler.setPromptTokensUsage(it->second.get_size());
+                            continue;
                         }
-                    } else {
-                        for (int i = 0; i < it->second.get_byte_size(); i++) {
-                            attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                        size_t attendedTokens = 0;
+                        if (it->second.get_element_type() == ov::element::Type_t::i64) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int64_t*>(it->second.data())[i];
+                            }
+                        } else if (it->second.get_element_type() == ov::element::Type_t::i32) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int32_t*>(it->second.data())[i];
+                            }
+                        } else {
+                            for (int i = 0; i < it->second.get_byte_size(); i++) {
+                                attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                            }
                         }
+                        handler.setPromptTokensUsage(attendedTokens);
                     }
-                    handler.setPromptTokensUsage(attendedTokens);
+                }
+            } else if (auto ints = std::get_if<std::vector<std::vector<int64_t>>>(&input)) {
+                size_t token_count_of_longest_document = 0;
+                int64_t pad_token = modelConfig["pad_token_id"].as<int64_t>();
+                size_t tokens = 0;
+                received_batch_size = ints->size();
+                for (auto i : *ints) {
```

Comment:
```suggestion
                for (const auto& document_tokens : *tokenized_documents) {
```
Use reference, otherwise the entire vector of integers is coped in every iteration
Use const since you do not need to modify it
Rename it for better readability